### PR TITLE
QT: Add 'copy full transaction details' option

### PR DESCRIPTION
**Pull request by ghost**
Adds feature from issue #7484

### CODE REVIEW

**Comment by ericshawlinux** [comment]
> No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.
That's a bit odd. My pull request compiled fine on every other system in the Travis CI build... so I'm not sure what to make of this.

**Comment by MarcoFalke** [comment]
No worries, it is a know issue unrelated to your changes.

**Comment by ericshawlinux** [comment]
ah, I see. what a relief

**Comment by laanwj** [comment]
Restarted travis, it should pass now.

**Comment by paveljanik** [comment]
ACK https://github.com/ericshawlinux/bitcoin/commit/fff0dcc4a42f384fa9e7778fa0241f3448454658

Minor NIT: empty '()' on addresses without labels. Can be tested beforehand...

For unconfirmed transaction:

```
3/3/16 08:05 Unconfirmed () mjHi7MpmjKzez9KY8zL6sSQVtBpqsaB9LF 2.60000000
```

4 confirms:

```
3/3/16 08:05 Confirming (4 of 6 recommended confirmations) () mjHi7MpmjKzez9KY8zL6sSQVtBpqsaB9LF 2.60000000
```

Confirmed:

```
3/3/16 08:05 Confirmed (6 confirmations) () mjHi7MpmjKzez9KY8zL6sSQVtBpqsaB9LF 2.60000000
```

With label:

```
3/3/16 08:05 Confirmed (6 confirmations) (Test label for #7628) mjHi7MpmjKzez9KY8zL6sSQVtBpqsaB9LF 2.60000000
```

The debug log contains this on OS X (the same warning as with Copy raw transaction:

```
2016-03-03 07:10:46 GUI: Data set on unsupported clipboard mode. QMimeData object will be deleted.
```

**Comment by paveljanik** [comment]
Off-topic here, but anyway: it would be nice to be able to add label to the address directly from the main window, in Recent transactions.

File: src/qt/transactiontablemodel.cpp [hunk 1]
```diff
@@ -609,6 +609,22 @@ QVariant TransactionTableModel::data(const QModelIndex &index, int role) const
...
+    case TxPlainTextRole:
+        {
+            QString details;
+            QDateTime date = QDateTime::fromTime_t(static_cast<uint>(rec->time));
+
+            details.append(date.toString("M/d/yy HH:mm"));
+            details.append(" ");
+            details.append(formatTxStatus(rec));
+            details.append(" (");
+            details.append(walletModel->getAddressTableModel()->labelForAddress(QString::fromStdString(rec->address)));
```
Comment (jonasschnelli) [inline src/qt/transactiontablemodel.cpp]: nit: what about only append L620 to L622 if `!rec->address.empty()`? I guess most users have couple of transaction without according address labels.

**Comment by jonasschnelli** [comment]
Tested ACK modulo nit.

Agree with @paveljanik about the `()` and the warning. Although the console-out warning is unrelated to this PR. If it can be fixed, it could be done in a different PR.

Binaries: https://bitcoin.jonasschnelli.ch/pulls/7628/

4 Examples 
```
3/3/16 13:19 Unconfirmed ()  -0.00003840
3/3/16 13:24 Unconfirmed (Yes! A testlabel.) mgY7FaksgbtrNPVJKzJBrKTG3Ayfqddc8R -1.00003840
2/22/16 09:27 Immature (6 confirmations, will be available after 101) () mr5C9MyoswwkTzFzwys6LfS5TdrX6u8xRo 0.39062500
2/22/16 09:27 Confirmed (143 confirmations) () mr5C9MyoswwkTzFzwys6LfS5TdrX6u8xRo 0.78125000
```

Two conceptual ideas for further improvements:

1. Would adding the transaction type make sense? In/Out/Mined? Or is the +/- in the balance enough, at least coinbase transaction are indistinguishable against "other" incoming txes?
2. If I select a row and hit Ctrl-C, currently, it does copy the cell-content I have last clicked (only date or only type, etc.). I think it would be nice to copy the "full details" if one presses Ctrl-C while having a row selected.

**Comment by ericshawlinux** [comment]
Yeah I could do that, I'll write no label if it's empty, and in out or mined on the end. Although the sign and the value does tell that, wouldn't a mined tx always be 50 or 25 btc?

**Comment by ericshawlinux** [comment]
oops, gonna have to push another fix :) I accidentally tested if the address was empty instead of the label. we can just squash these all together I think.

File: src/qt/transactiontablemodel.cpp [hunk 2]
```diff
@@ -609,6 +609,34 @@ QVariant TransactionTableModel::data(const QModelIndex &index, int role) const
...
+            details.append(" ");
+            details.append(formatTxStatus(rec));
+            details.append(". ");
+            if(!formatTxType(rec).isEmpty()) {
+                details.append(formatTxType(rec));
+                details.append(" ");
+            }
+            if(!rec->address.empty()) {
+                if(txLabel.isEmpty())
+                    details.append("(no label) ");
```
Comment (jonasschnelli) [inline src/qt/transactiontablemodel.cpp]: We should use `details.append(tr("(no label)")+" ");` here to allow translations.

**Comment by jonasschnelli** [comment]
Can you squash your commits?

**Comment by ericshawlinux** [comment]
Ok

File: src/qt/transactiontablemodel.cpp (hunk 2 above)
Comment (ericshawlinux) [inline src/qt/transactiontablemodel.cpp]: actually, the tr() call isn't doing what I expect for some reason. I'll have to investigate the reason next and I'll resquash now that I know how.

**Comment by jonasschnelli** [comment]
Tested ACK b51ed4036e157a116414f53ac8da78c6b35bf041.

**Pull request merged**